pkg/client/cache/store.go

```diff
@@ -76,6 +81,57 @@ func (c *cache) Add(obj interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.items[key] = obj
+	c.updateIndices(obj)
```

Comment:
@smarterclayton - its not rebuilding the entire index.  your point on prev object is needed, will update.